### PR TITLE
Add balance to the blockchain address GQL type

### PR DESCRIPTION
**Pull request by IvanIvanoff**
## Changes

Rework the historical balance selector to not explicitly require `infrastructure` if `slug` is provider - it will be deduced automatically.

Add `balance(slug: "<slug>")` field on the blockchain address type. It can be used either when querying a single address or when using it with watchlists (it will use dataloader for faster load here).
Request:
```graphql
{
  watchlist(id:1173){
    listItems{
      blockchainAddress{
        address
        commentsCount
        id
        ethBalance: balance(selector: {slug: "ethereum"})
        sanBalance: balance(selector: {slug: "santiment"})
      }
    }
  }
}
```

Response:
```json
{
  "data": {
    "watchlist": {
      "listItems": [
        {
          "blockchainAddress": {
            "address": "0x04291180af677efd464432138d2a5d766a3898d4",
            "commentsCount": 0,
            "ethBalance": 0.005463449071446587,
            "id": 1,
            "sanBalance": 0
          }
        },
        {
          "blockchainAddress": {
            "address": "0x00e6a95f6f438cbe873d0465e1a2533052e155f0",
            "commentsCount": 0,
            "ethBalance": 0.03776633905,
            "id": 2,
            "sanBalance": 0
          }
        },
        {
          "blockchainAddress": {
            "address": "0x003b9fb45362194a35a30cf076c937b5f483dc73",
            "commentsCount": 0,
            "ethBalance": 0.022446721,
            "id": 3,
            "sanBalance": 10995.749999990001
          }
        }
      ]
    }
  }
}
```

## Checklist:

- [x] I have performed a self-review of my own code
- [x] I have made corresponding changes to the documentation
- [x] I have tried to find clearer solution before commenting hard-to-understand parts of code
- [x] I have added tests that prove my fix is effective or that my feature works

### CODE REVIEW

File: lib/sanbase/blockchain_address/blockchain_address.ex [hunk 1]
```diff
@@ -26,6 +26,13 @@ defmodule Sanbase.BlockchainAddress do
     end
   end
 
+  def to_internal_format(address) do
+    case Regex.match?(~r/^0x([A-Fa-f0-9]{40})$/, address) do
```
Comment (tspenov) [inline lib/sanbase/blockchain_address/blockchain_address.ex:30]: can you add a comment here - why this type is downcased but others not ? I guess this should be ethereum addresses :? but is it matching only those ?

File: lib/sanbase/clickhouse/historical_balance/fetchers/eth_balance.ex [hunk 2]
```diff
@@ -27,7 +27,7 @@ defmodule Sanbase.Clickhouse.HistoricalBalance.EthBalance do
 
   @impl Sanbase.Clickhouse.HistoricalBalance.Behaviour
   def assets_held_by_address(address) do
-    {query, args} = current_balances_query(address)
+    {query, args} = asset_held_by_address_query(address)
```
Comment (tspenov) [inline lib/sanbase/clickhouse/historical_balance/fetchers/eth_balance.ex:30]: wow, why was it fetching current balance in `assets_held_by_address` ?

Comment (IvanIvanoff) [reply lib/sanbase/clickhouse/historical_balance/fetchers/eth_balance.ex:30]: `assets_held_by_address` API returns the assets held *and* the current balance of that asset.

It was correct, the function name was just not very proper. In order to see if the given address has ether in it (the only asset this module handles) it just checked the current balance - if there is such record it is returned along with the balance.

File: lib/sanbase/blockchain_address/blockchain_address.ex (hunk 1 above)
Comment (IvanIvanoff) [reply lib/sanbase/blockchain_address/blockchain_address.ex:30]: I'll add.

**Pull request merged**